crypto/bio/b_print.c

```diff
@@ -635,7 +635,7 @@ fmtfp(char **sbuffer,
             fvalue = tmpvalue;
     }
     ufvalue = abs_val(fvalue);
-    if (ufvalue > ULONG_MAX) {
+    if (ufvalue > (LDOUBLE)ULONG_MAX) {
```

Comment:
I don't think this is a proper solution.  `sizeof(long double) == sizeof(double)` on many platforms and the problem remains.

`ULONG_MAX + 1` can be exactly represented as a double.  Unfortunately, `ULONG_MAX + 1` isn't calculable directly.

I'll have a think about solutions....

Reply:
Yes, I was just checking that. At least on my arm it's the same.

Reply:
I think changing it from > to >= will actually fix it, the implicit conversion seems to change it to the value we want.

Reply:
But that also doesn't actually fix anything. I think we just need to check for it being >= 1<<53, or something like that.

Reply:
But it gets cast to an integer, and the the integer part gets subtracted again ... So maybe the patch will actually do the correct thing. I guess I currently don't understand the code good enough.

Reply:
> But that also doesn't actually fix anything. I think we just need to check for it being >= 1<<53, or something like that.

Changing the comparison to 1<<53 seems reasonable.

It's not quite a fix: 1<<60 is exactly representable by both a double and a long integer.  Still, I think it's good enough for now.